grumpy-runtime-src/runtime/builtin_types.go

```diff
@@ -808,12 +812,58 @@ func builtinSum(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	} else {
 		result = NewInt(0).ToObject()
 	}
-	raised := seqForEach(f, args[0], func(o *Object) (raised *BaseException) {
-		result, raised = Add(f, result, o)
-		return raised
-	})
-
+	if result.typ == IntType {
```

Comment:
There are checks for IntType and FloatType to optimize this cases. If this becomes a pattern the code will be messed very soon.

I do think that it should me moved into `Add()` instead. Agree?